src/ddmd/parse.d

```diff
@@ -4795,6 +4798,202 @@ final class Parser(AST) : Lexer
            ddmd.errors.deprecation(loc, "instead of C-style syntax, use D-style syntax '%s%s%s'", t.toChars(), sp, s);
     }
 
+    private static template ParseForeachArgs(bool isStatic, bool isDecl)
```

Comment:
This determines the signature of parseForeach depending on `isStatic` and `isDecl`. (It returns an AliasSeq, so needs to be a template.) I'll document it.